routes/projects/update.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * PUT /projects/:username/:projectName
+ * Creates a new project for the given user. A project object is sent in
+ * the form:
+ * {
+ *   name: <string> required,
+ *   username: <string> required,
+ *   isPublic: <boolean> required,
+ *   publishedAt: <dataTime> optional,
+ *   publishLink: <string> optional,
+ *   files: [ <FileObj1>, <FileObj2>... ] required
```

Comment:
I think this API's view of files associated with a project has to relate to what's on the server, which is to say, I don't think a client should be able to specify it, only create, delete, and rename files for a project.  Since this payload won't include those files, I'd suggest you remove this, and just use `[]` until the files themselves are uploaded.